spec/support/factories/weapons.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+FactoryBot.define do
+  factory :weapon do
+    game
+
+    name { 'Dwarven War Axe' }
+
+    trait :with_matching_canonical do
+      association :canonical_weapon,
+                  factory: :canonical_weapon,
+                  strategy: :create
+
+      name { canonical_weapon.name }
```

Comment:
We don't really need to define this here since the default name for a canonical weapon is the same as that for an in-game weapon.